src/Cli/dotnet/Commands/Run/VirtualProjectBuildingCommand.cs

```diff
@@ -1414,7 +1414,7 @@ public static void WriteProjectFile(
 
             writer.WriteLine($"""
                   <ItemGroup>
-                    <Compile Include="{EscapeValue(targetFilePath)}" />
+                    <Compile Condition="'$(EnableDefaultCompileItems)'!='true'" Include="{EscapeValue(targetFilePath)}" />
```

Comment:
I think you will need to update some existing RunFileTests which verify the contents of the whole virtual project (the tests are prefixed with `Api_` iirc).

Let's also use spaces around the operator:
```suggestion
                    <Compile Condition="'$(EnableDefaultCompileItems)' != 'true'" Include="{EscapeValue(targetFilePath)}" />
```

Consider also adding a comment here before the writer.WriteLine, explaining why the condition is needed.